Four texts in foundations added

noahschoem opened the pull request:
I added four texts in foundations that I have used and think are good texts to recommend.

Kunen could arguably go in the advanced section, since its material is sufficiently more difficult than Halmos or Ebbinghaus/Flum/Thomas. However, I believe that a third year math major or sufficiently advanced second year math major at UChicago could navigate Kunen perfectly well, and its contents are foundational enough to modern set theory that they really shouldn't be considered advanced foundations material.

I'm not thrilled to recommend Chang & Keisler; it's an excellent introduction to model theory, but isn't really a good representation of the modern field.  I'll ask some colleagues for better recommendations.

nschoem committed:
Additional foundations texts added in intermediate and advanced sections on set theory, with one model theory text in advanced.

nschoem committed:
Chang & Keisler really belongs in intermediate

ystael commented on intermediate/foundations.md:
```diff
@@ -36,6 +49,14 @@ about what's going on and less unmotivated symbol-pushing than in E/F/T.  The fl
 is, the constructions may or may not be epsilon less precise.  I'm not a logician; if you are,
 write some reviews so I can replace these lousy ones!
 
+#### Chang & Kiesler, *Model Theory*
```
Spelling -- I believe this is 'Keisler'?

noahschoem replied:
Oops, yes it is Keisler.

ystael reviewed (commented):
You've given good orienting summaries for each of these books, but except for Chang and Keisler there's not a lot of personal perspective on why they are good or not-so-good sources. Do you have any more opinions of that kind to add, or anything that might help a student not educated in foundations figure out why they might want to look at these books?

noahschoem committed:
Update foundations.md

nschoem committed:
Substantial rewrite of Kunen summary
Summary needed substantial rewrite to provide my personal perspective on Kunen, in addition to why someone might want to read it.

nschoem committed:
Further words on Jech and Handbook
...to add personal opinions and why/why nots to both summaries

noahschoem commented:
I have reworked the summary of Kunen, and added to the summaries for Jech and the Handbook, to include my perspectives and some guidance on the summaries.  I may add more for the Handbook.

nschoem committed:
Further exposition and recommendations for Handbook
Now reflects that Chapter 1 is a great historical overview to modern set theory; also, that the rest is incredibly dense high-level material.
